tests/foreman/ui/test_user.py

```diff
@@ -289,3 +289,73 @@ def test_positive_update_orgs(session):
             set(user['organizations']['resources']['assigned']) ==
             set(org_names)
         )
+
+
+@tier2
+def test_positive_create_product_with_limited_user_permission(session, test_name,
+                                                              module_org, module_loc):
+    """A user with all permissions in Product and Repositories should be able to
+    create a new product
+
+    :id: 534a16f9-2d66-4fa1-aa5b-560f00eb4f67
+
+    :expectedresults: User successfully creates new product
+
+    :caseLevel: Component
+
+    :CaseImportance: High
+
+    :BZ: 1771937
+    """
+    role_name = gen_string('alpha')
+    role_description = gen_string('alpha')
+    resource_type = 'Product and Repositories'
+    permissions = ['view_products', 'create_products', 'edit_products',
+                   'destroy_products', 'sync_products', 'export_products']
+    username = gen_string('alpha')
+    password = gen_string('alpha')
+    product_name = gen_string('alpha')
+    product_label = gen_string('alpha')
+    product_description = gen_string('alpha')
+
+    with session:
+        session.role.create({
```

Comment:
you can fast this test with making dependent steps using API call instead everything over UI. 

@latran what do you think?

Reply:
@omkarkhatavkar, I was thinking the same.  I tried to implement it last week without much luck and it was slowing down my progress so I did it this way.  I can try to implement it again.

Reply:
@latran take a look at `tests/foreman/ui/test_contentview.py::test_positive_non_admin_user_actions` for example of UI test that uses API to create role and user.

Reply:
@mirekdlugosz thanks!  I do have something working at the moment.  The part I tripped up was checking the permissions in the role because the api call was different than what I expected.  I'll update this PR as soon as I can.